### PR TITLE
Feature branch

### PR DESCRIPTION
**Pull request by lirad**
In this project, we created two sorting algorithms Bubble Sort, to arrays by number and Bubble Sort By, to sort strings:
- Created Bubble Sort method
- Created Bubble Sort By method
- Output the results with test cases

### CODE REVIEW

File: bubble_sort.rb [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+      array[index + 1] = a
+      array[index] = b
+      swap_counter += 1
+    end
+    # Stop the loop if array is sorted
+    break if swap_counter.zero?
+  end
+  yield
+end
+
```
Comment (bertil291utn) [inline bubble_sort.rb]: ## Bubble_sort
- This function should return an array. Follow this [example](https://github.com/bertil291utn/bubble-sort-project/blob/master/bubble_sort_algorithm.rb)
- Then try to check this test
```ruby
unsorted = (1..10).to_a.reverse!
bubble_sort(unsorted))
# should be equal to  
unsorted.sort
```

File: bubble_sort.rb [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+      array[index + 1] = a
+      array[index] = b
+      swap_counter += 1
+    end
+    # Stop the loop if array is sorted
+    break if swap_counter.zero?
+  end
+  yield
+end
```
Comment (bertil291utn) [inline bubble_sort.rb]: ## Bubble_sort_by
- [x] This function should return an array. Follow this [example](https://github.com/bertil291utn/bubble-sort-project/blob/master/bubble_sort_algorithm.rb)
- [x] Try testing this scenario

```ruby
unsorted=%w[hi hello hey]
bubble_sort_by(unsorted) { |left, right| left.length - right.length })
# should be equal to 
%w[hi hey hello]
```

File: bubble_sort.rb [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+# bubble_sort_by(["hi","hello","hey"])
+bubble_sort_by(string_arr) { puts "Sorted String: #{string_arr}" }
```
Comment (bertil291utn) [inline bubble_sort.rb]: ## Debugging instructions
- No debugging instructions in committed code (as discussed [here](https://questions.microverse.org/t/debugging-instructions-code/346)) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/section/debugging_requirement.png) from Pathwright)
  - [ ] Once you finished you must remove all debugging

File: bubble_sort.rb [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+      swap_counter += 1
+    end
+    # Stop the loop if array is sorted
+    break if swap_counter.zero?
+  end
+  yield
+end
+
+# bubble_sort([4,3,78,2,0,2])
+bubble_sort(integer_arr) { puts "Sorted Integer: #{integer_arr}" }
```
Comment (bertil291utn) [inline bubble_sort.rb]: ## Debugging instructions
- No debugging instructions in committed code (as discussed [here](https://questions.microverse.org/t/debugging-instructions-code/346)) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/section/debugging_requirement.png) from Pathwright)
  - [ ] Once you finished you must remove all debugging

**Review by bertil291utn — CHANGES_REQUESTED** [review]
# **Status:** Changes requested 🟠 
<p align="center"> image <br><strong>Not yet</strong> </p>

# Hey guys, you've been doing a great job, and also I hope you enjoyed coding.

### Your work is amazing you master it GitHub Flow you have a professional README file. 

However, in order to all possible scenarios, you required to do these changes, listed below:

- [x] Check the inline comments

### Summary

-  Check all the possible scenarios listed in the inline comments 

***

Guys work in the requested changes above and request another review.
Good luck. 

> “Never stop, never conform until what is good becomes even better, and what’s better becomes excellent.”

File: bubble_sort.rb [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+      array[index + 1] = a
+      array[index] = b
+      swap_counter += 1
+    end
+    break if swap_counter.zero?
+  end
+  yield
+  array
+end
```
Comment (tirthajyoti-ghosh) [inline bubble_sort.rb]: You are calling `yield` inside the `bubble_sort` method where you don't need it. Please remove it.

File: bubble_sort.rb [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+bubble_sort(integer_arr) { puts "Sorted Integer: #{integer_arr}" }
```
Comment (tirthajyoti-ghosh) [inline bubble_sort.rb]: `bubble_sort` method should not accept a block. Please remove this as well.

File: bubble_sort.rb [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+
+      array[index + 1] = a
+      array[index] = b
+      swap_counter += 1
+    end
+    break if swap_counter.zero?
+  end
+  yield
+  array
+end
```
Comment (tirthajyoti-ghosh) [inline bubble_sort.rb]: You are calling the `yield` in the wrong place. The `yield` keyword invokes the block that is passed to the `bubble_sort_by` method when called.

File: bubble_sort.rb [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+
+bubble_sort_by(string_arr) { puts "Sorted String: #{string_arr}" }
```
Comment (tirthajyoti-ghosh) [inline bubble_sort.rb]: This method should accept a condition inside the block that will determine "how" to sort the array passed as a parameter.

**Review by tirthajyoti-ghosh — CHANGES_REQUESTED** [review]
# Changes required 🛑 

## Impressive progress so far 💪! Thanks for making all the changes 👏.

### However, we need to work on some changes before we can merge.

Please make the correct use of `yield`.

Check the comments below.

**Review by mikenath223 — APPROVED** [review]
### CONGRATULATIONS  👏👏👏

This project meets the requirements for approval.

**STATUS: APPROVED ✅**


Don't forget to fill the project completion form👍
Good luck with the next project.

HAPPY CODING👨‍💻

**Pull request merged**